### ISSUE TITLE
Rename/alias R.of --> R.listOf

### ISSUE THREAD

**Issue by vitalets** [ramda/ramda#2502]
For me personally [`R.of`](http://ramdajs.com/docs/#of) has a confusing name. Every time I meet it in code I spend extra cognitive payload on it. Unlike `R.objOf` that is rather clear.

What do you think about renaming/aliasing it to `R.listOf` / `R.toArray`?

I've read that ramda [should be 100% without aliases](https://github.com/ramda/ramda/pull/574#issuecomment-64129603). I understand that this change can break existing codebase. But, from the other side, I believe that clear function naming is an important point. Especially for newcomers.

**Comment by CrossEye** [comment]
The chief reason for this function is to work with the [Fantasyland] [Applicative specification][fa], which gives us for instance, `Maybe.of(42) //=> Maybe.Just(42)`.  So we can't really change the name.  And as you note, we have accepted a strong no-alias policy. 

I agree that the name is not wonderful.  But I don't see that there's much to do.


  [fa]: https://github.com/fantasyland/fantasy-land#applicative

**Comment by GenCodeDev** [comment]
Apologies for being a bit off-topic, but a quick question... @CrossEye I see your reference to **Maybe** in your comment above.  I'm confused.  Does Ramda have a "Maybe" construct?  I'm just learning, but it seemed to be one thing that was lacking.  What did I miss?

Thanks.

**Comment by Bradcomp** [comment]
@CrossEye I was considering writing something similar to what you put, but then I thought about it, and our `of` can't handle anything other than Arrays. It doesn't dispatch, and of course it can't dispatch given its current signature. 

It would be nice to have an `of` function that actually did do some dispatching.

```

**Comment by CrossEye** [comment]
@Bradcomp: I forgot that we had this issue.  I agree that we need to get to that and pretty soon.  I have a few Ramda things on my plate, but will move this up the list.  I agree that we should do something like that formulation.  (And then @vitalets could get his wish too!)

**Comment by CrossEye** [comment]
@GenCodeDev: No, Ramda does not include `Maybe` or other similar types.  There was a now-dead Ramda-fantasy project for such things.  But we didn't put the effort in to keep it updated.  And there were many good libraries out there for this.

I've [described][so] Ramda as not supplying new types, simply operating on built-in, and often custom, types.  But it works well with (a subset) of the FantasyLand specification, so you can use it with many types, such as most `Maybe` implementations.


  [so]: https://stackoverflow.com/a/33130194/1243641

**Comment by GingerPlusPlus** [comment]
Is the `arrayOf` function actually needed at all? It can be easily implemented as `unary(`[`list`](https://github.com/ramda/ramda/wiki/Cookbook#create-a-list-function)`)`,
or just `x => [x]`.

Similary, `pair = curry(binary(list))`.

**Comment by CrossEye** [comment]
It needs to be fixed to dispatch properly.  I don't have the time right now like digging to find why we had problems doing so.  But we need to address them, so that we can work properly with Applicatives.

**Comment by customcommander** [comment]
> @CrossEye I was considering writing something similar to what you put, but then I thought about it, and our `of` can't handle anything other than Arrays. It doesn't dispatch, and of course it can't dispatch given its current signature.
> 
> It would be nice to have an `of` function that actually did do some dispatching.
> 
> ```js
> of(Array, 42) //> [42]
> of(Maybe, 42) //> Just(42)
> of(Either, 42) //> Right(42)
> ```
> 
> and so on.

I quite like this. @CrossEye should be simply rewrite `R.of` to work as @Bradcomp suggested for `v1.0`. Quite a substantial breaking change but I reckon we should try to ship as many of them in `v1.0` since that would be our first semver compliant release.

**Comment by CrossEye** [comment]
@customcommander: Agreed.  And then we can get the OP's `listOf` as a trivial `of(Array)`.
